src/parser.rs

```diff
@@ -62,6 +76,14 @@ pub struct BasicParseError<'i> {
     pub location: SourceLocation,
 }
 
+impl<'i> fmt::Display for BasicParseError<'i> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{}", self.kind.description())
+    }
+}
+
+impl<'i> Error for BasicParseError<'i> {}
```

Comment:
Why did you not implement `description` here?

Reply:
The documentation for `Error` says: "This method is soft-deprecated. Although using it won’t cause compilation warning, new code should use Display instead and new impls can omit it."

Reply:
Please implement it anyway, calling `BasicParseErrorKind::description`.